File: python/raindrops/raindrops.py
def convert(number: int) -> str:
    rain_sound: str = ""

    rain_sound += "Pling" if number % 3 == 0 else ""
    rain_sound += "Plang" if number % 5 == 0 else ""
    rain_sound += "Plong" if number % 7 == 0 else ""

    return rain_sound or str(number)

File: ruby/raindrops/raindrops.rb
module Raindrops
  def self.convert(number)
    rain_sound = { 3 => "Pling", 5 => "Plang", 7 => "Plong" }.inject("") do |drops, (key, value)|
      drops += value if number % key == 0
      drops
    end

    if rain_sound.empty?
      number.to_s
    else
      rain_sound
    end
  end
end
